### PR TITLE
Implement `unchunks` (inverse of `chunks`)

### PR DESCRIPTION
**Pull request by armanbilge**
This seemed a strange omission to me, but maybe there's a good reason for it?

### CODE REVIEW

**Comment by mpilquist** [comment]
There's already `unchunk` which (unfortunately) does something different and we never found a name that we were happy with.

**Comment by djspiewak** [comment]
The only concern I see is the fact that `unchunk` is a function which you almost never want to touch, while `unchunks` is a perfectly normal thing that you'll do all the time. :-P

**Comment by armanbilge** [comment]
> The only concern I see is the fact that unchunk is a function which you almost never want to touch

If this is true, can we deprecate and rename `unchunk`? Would deprecation be enough warning when writing new code, _is this really the method you want to use?_

**Comment by mpilquist** [comment]
I'd be fine with deprecating `unchunk` and giving its replacement a scarier name like `destroyChunks` or `unitaryChunks` or something?

**Comment by armanbilge** [comment]
It's a very obtuse method actually, because the change is happening under-the-hood right? What are its use cases? I would guess maybe when you don't want eager evaluation, so you don't lose a whole chunk on error? Could the name reflect something about that?

**Comment by djspiewak** [comment]
Yeah I've used O.G. `unchunks` like… once… The use-case is when you have a downstream `flatMap` which produces considerably more data on a per-element basis. If you *don't* unchunk in that case you can have memory issues.

**Comment by armanbilge** [comment]
Another idea: if the OG `unchunk` use-case is so rare, why don't we just suggest replacing with `.chunkLimit(1).unchunks`?

**Comment by djspiewak** [comment]
> Another idea: if the OG unchunk use-case is so rare, why don't we just suggest replacing with .chunkLimit(1).unchunks?

This is superb.

**Comment by mpilquist** [comment]
With current implementation of `chunkLimit`, that will be a bit different (compare expanded form of `uncons1` with `unconsLimit(1)`) -- we should probably fix that

**Comment by armanbilge** [comment]
@mpilquist excuse my density, what exactly is the difference?

**Comment by mpilquist** [comment]
The special casing which avoids the cons call when chunk size is equal to 1

**Comment by armanbilge** [comment]
I still don't see how that makes a difference, other than performance? Since `cons` will ignore an empty chunk.

**Comment by mpilquist** [comment]
Only performance, though if full expansion is literally only the call to cons and doesn't actually change anything, then disregard

**Comment by armanbilge** [comment]
The special case avoids the `cons` _and_ the `drop`/`splitAt`. I can make this change?

**Comment by armanbilge** [comment]
But btw, how important is performance here anyway? In @djspiewak's use-case if the stream is about to get slammed by flatMapping to big chunks does this really matter?

File: core/shared/src/main/scala/fs2/Stream.scala [hunk 1]
```diff
@@ -2588,6 +2588,7 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     * res0: List[Chunk[Int]] = List(Chunk(1), Chunk(2), Chunk(3), Chunk(4), Chunk(5), Chunk(6))
     * }}}
     */
+  @deprecated("Use .chunkLimit(1).unchunks instead.", "3.0.7")
```
Comment (mpilquist) [inline core/shared/src/main/scala/fs2/Stream.scala:2591]: This should have introduced a bunch of deprecation warnings in the test suite. Can we address those too?

Comment (armanbilge) [reply core/shared/src/main/scala/fs2/Stream.scala:2591]: On it.

**Review by Daenyth — APPROVED** [review]
I've wanted this since Segment was still a thing

**Pull request merged**